x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/EsqlSession.java

```diff
@@ -313,7 +313,7 @@ private <T> void preAnalyze(
             // First resolve the lookup indices, then the main indices
             preAnalyzeLookupIndices(
                 preAnalysis.lookupIndices,
-                fieldNames,
+                Set.of("*"), // Current LOOKUP JOIN syntax does not allow for field selection
```

Comment:
I think the comment is insufficient; I believe this is a good hack but should be improved later; in a query like
```
FROM idx | LOOKUP JOIN lookup_idx ON lookup_field | KEEP another_lookup_field
```
we do not have to ask for all existing field names. We just need `another_lookup_field`.

In fact, I think this will make it so that we ask the `LookupFromIndexService` to fetch **all** fields from the index, meaning that the performance of this operation will depend on the number of fields in the lookup index. In scenarios where that index has, say, 6000 fields (not completely unheard of), that may be a serious performance drag and could cause memory issues.

To be more precise, I think we'll have to augment`fieldNames`. We should be able to walk the tree from the top, _stopping_ at the first `LOOKUP JOIN` we see - and then determining that the missing fields _so far_ are the only candidates that may come from the lookup index. This means that the output of `fieldNames` shouldn't be a single `Set<String>` - but, instead, a `Set<String>` for the main index and a `List<Set<String>>` corresponding to the missing fields we might obtain from the lookup indices.

Reply:
The comment in the followup work is:
>EsqlSession.fieldNames does not handle lookup references that are also mentioned in aliases (erases them)

So we want to fix this bug, and I have a reasonable idea how.